reactive/openstack.py

```diff
@@ -203,11 +203,46 @@ def _validate_loadbalancer_request(request: "LBRequest") -> "LBResponse":
         if hc.path and hc.protocol.value not in ("http", "https"):
             error_fields["hc[{}].path".format(i)] = "Only valid with http(s) protocol"
 
+    remote_port: Optional[int] = None
     config = hookenv.config()
     lb_port = int(config["lb-port"])
-    remote_port = request.port_mapping.get(lb_port)
-    if request.backends and not remote_port:
-        error_fields["port_mapping"] = f"Not configured for {lb_port}"
+    if request.port_mapping:
+        if len(request.port_mapping) > 1:
+            hookenv.log(
+                "Multiple port mappings are specified in the request.\n"
+                + f"{request.port_mapping=}",
+                hookenv.WARNING,
+            )
+
+        # NOTE(Hue): For compatibility reasons, we only respect a single
+        # port mapping. we will first try to find the port mapping for the
+        # configured `lb-port`.
+        # If that fails, we will use on from the port_mapping dict.
+
+        # Try finding the port mapping for the configured `lb-port`.
+        remote_port = request.port_mapping.get(lb_port)
+
+        # If the port mapping for the configured `lb-port` is not found,
+        # use one from the port_mapping dict.
+        if not remote_port:
+            hookenv.log(
+                f"No port mapping found for the configured `lb-port` ({lb_port})",
+                hookenv.WARNING,
+            )
+            lb_port, remote_port = list(request.port_mapping.items())[0]
```

Comment:
i dunno why this looks a bit cleaner.  There's just not a great way of doing this in python
```suggestion
            lb_port, remote_port = next(iter(request.port_mapping.items())
```